src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java

```diff
@@ -48,6 +53,12 @@ public Claim save(
 
         countyCourtJudgmentRule.assertCountyCourtJudgementCanBeRequested(claim, issue);
 
+        if (countyCourtJudgment.getPaymentOption() == PaymentOption.INSTALMENTS) {
+            claimantRepaymentPlanRule.assertClaimantRepaymentPlanIsValid(claim,
+                countyCourtJudgment.getRepaymentPlan().orElse(null));
```

Comment:
As in admissions CCJ, `countyCourtJudgment` may have payment plan from defendant response or claimant alternate or court generated based on the determination flow. The name `assertClaimantRepaymentPlanIsValid` and class name `ClaimantRepaymentPlanRule` does not seems right.

Reply:
@ellie-harrison I need to double with Marc as I am not sure we expected to have this validation for CCJ by admission, but good point :-) I believe the scope of this user story should only cover DETERMINATION. 

Also let's have a chat about where is the most appropriate place for this validation. Since it is not CCJ centric, nor SETTLEMENT centric and it should also apply for REFER_TO_JUDGE it feels to me we should run this logic earlier in the claimant response controller.